packages/next/src/executors/build/schema.json

```diff
@@ -43,5 +43,5 @@
       "default": true
     }
   },
-  "required": []
+  "required": ["root", "outputPath"]
```

Comment:
This is a breaking change because if people didn't have these options before, it would have worked, but now it won't.

What happened when this wasn't there before?

Is there a way to set these during a migration so that we break less people?

Reply:
ah, very good point..

> What happened when this wasn't there before?

1. `root` is the more important one out of the two. as we've seen with our nx-dev project a few weeks ago, without it, nx would just error out 😔 :
```
> nx run nx-dev:serve 
The "path" argument must be of type string. Received undefined
```

2. as for `outputPath` - without it, nx won't put the build artefacts into `dist` at all.. it adds it to the root of the workspace into a new directory called `undefined`..

so to be fair, missing either of those would result in quite broken behavior.. 

3. missing `buildTarget` (this is on the serve executor) will also simply error out:
```
> nx run nx-dev:serve 
Cannot read property 'split' of undefined
```

---

as for migrating, i'm actually now thinking that there *is* a way to keep these values optional:
- for missing `root` we can just default to using the project `root`
- similarly, for missing `outputPath`, we can prefix the project `root` with `dist`
- `buildTarget` is a bit trickier. i guess we *could* default to `<project name>:build`. not sure if project name is easily available for us to use in the implementation, but i imagine it is.

i guess the question is which solution we prefer - sticking with the breaking change and implementing above as a migration? or, baking these defaults into the implementation? having said that, perhaps there's a hybrid approach, where we bake the defaults into the implementation for now, and then do the breaking change in time for Nx 13? 
this also isn't that urgent of a fix, we could wait for Nx 13 with this change + a migration. 

what are your thoughts?

Reply:
We discussed it, and those properties lead to errors at the moment so they basically are required. Thanks @kirjai

Reply:
Jason and I caught up outside of this thread and decided that since missing these properties out would either error out or result in broken behavior - that there's no need to mark this as a breaking change